viewer-prototype/src/browser/trace-explorer/trace-explorer-widget.tsx

```diff
@@ -12,6 +12,7 @@ import { EditorManager, EditorOpenerOptions } from '@theia/editor/lib/browser';
 import URI from '@theia/core/lib/common/uri';
 import { Experiment } from 'tsp-typescript-client/lib/models/experiment';
 import { ExperimentManager } from '../../common/experiment-manager';
+import { TraceViewerWidget } from '../trace-viewer/trace-viewer';
```

Comment:
Just highlighting this is adding a circular dependency between trace-viewer and trace-explorer-widget. It is definitely not a good practice in general, even though it works here.

With the split in packages and vscode extension, this is the kind of interaction that may well be re-architectured (how signals are sent between components, etc), so you don't have to try to fix it just yet. But could you add a FIXME above this line, just so we can come back to this eventually?

Reply:
Thank you for sharing this.

Reply:
Added a `FIXME` comment. Thanks.